### PR TITLE
Only show browser comparison pages in DEV mode

### PR DESCRIPTION
**Pull request by craigcook**
**Note:** we should only merge this **if** we need to deploy to production **before** we get a go-ahead on the comparison pages. We can also just wait to push prod.

## Description
The browser comparison pages are finished, but it was only after they merged that I learned we're still waiting for a final go/no-go from leadership before publishing them, and we won't get word until tomorrow (25 June) at the earliest. 

Since the pages already merged we're effectively blocked from doing a production push until we have permission to relaunch the comparison pages. However, l10n is already in progress so we need to make sure the pages are still available on dev/demo, so I don't want to revert the commit or redirect the URLs. I considered putting the pages behind a switch but I think that would entail routing them through a view (don't think we can do switches in urls.py) and this dev mode rule was simpler.

tl;dr: this is a quick-and-dirty hack just to unblock us from pushing production. We might not even need it.

## Testing
http://localhost:8000/firefox/browsers/compare/ (and sub-pages)

- [ ] Pages should load normally when `DEV=True`
- [ ] Pages should redirect to /firefox when `DEV=False`

### CODE REVIEW

**Comment by craigcook** [comment]
Looks like we're gonna need this after all!
